pss/outbox/outbox_test.go

```diff
@@ -158,10 +168,62 @@ func newTestMessage(num byte) *message.Message {
 	}
 }
 
-var testOutboxMessage = outbox.NewOutboxMessage(&message.Message{
-	To:      nil,
-	Flags:   message.Flags{},
-	Expire:  0,
-	Topic:   message.Topic{},
-	Payload: nil,
-})
+func TestMessageRetriesExpired(t *testing.T) {
+	failForwardFunction := func(msg *message.Message) error {
+		return errors.New("forward error")
+	}
+
+	// We are going to simulate that 5 minutes has passed with a mock Clock
+	duration := 5 * time.Minute
+	now := time.Now()
+	mockClock := clock.NewMock(now)
+	testOutbox := outbox.NewMock(&outbox.Config{
+		NumberSlots:  1,
+		Forward:      failForwardFunction,
+		MaxRetryTime: &duration,
+		Clock:        mockClock,
+	})
+
+	testOutbox.Start()
+	defer testOutbox.Stop()
+
+	msg := testOutbox.NewOutboxMessage(&message.Message{
+		To:      nil,
+		Flags:   message.Flags{},
+		Expire:  0,
+		Topic:   message.Topic{},
+		Payload: nil,
+	})
+
+	err := testOutbox.Enqueue(msg)
+	if err != nil {
+		t.Errorf("Expected no error enqueueing, instead got %v", err)
+	}
+
+	numMessages := testOutbox.Len()
+	if numMessages != 1 {
+		t.Errorf("Expected one message in outbox, instead got %v", numMessages)
+	}
+
+	mockClock.Set(now.Add(duration / 2))
```

Comment:
Would you not need the same retry as below to verify here?

Reply:
Well, bellow I'm waiting for something to happen, in this case I should wait for something not to happen. It is hard to do that because in the case everything works fine I should wait the whole time, delaying the tests execution. In the last case, even though there are 10 iterations waiting, in the normal case, it will take only 10 milliseconds to finish.
What I can do is wait a couple of iterations.